pkg/oomstore/join.go

```diff
@@ -73,31 +71,24 @@ func (s *OomStore) ChannelJoin(ctx context.Context, opt types.ChannelJoinOpt) (*
 }
 
 // Join gets point-in-time correct feature values for each entity row.
-// The method is similar to Join, except that both input and output are files on disk.
+// The method is similar to Join, except a input files on disk.
 // Input File should contain header, the first two columns of Input File should be
 // entity_key, unix_milli, then followed by other real-time feature values.
-func (s *OomStore) Join(ctx context.Context, opt types.JoinOpt) error {
-	ctx, cancel := context.WithCancel(ctx)
-	defer cancel()
-
+func (s *OomStore) Join(ctx context.Context, opt types.JoinOpt) (*types.JoinResult, error) {
```

Comment:
In this way, Go SDK will behave differently from other SDKs, is that expected?

Reply:
This makes go SDK more flexible. Because other SDKs need to go through the oomagent layer of wrappers.
As for the inconsistent behavior, I'm not quite sure of the scope of the impact, we can discuss it.